internal/archiver/archiver.go

```diff
@@ -208,7 +208,7 @@ func (arch *Archiver) loadSubtree(ctx context.Context, node *restic.Node) *resti
 
 // SaveDir stores a directory in the repo and returns the node. snPath is the
 // path within the current snapshot.
-func (arch *Archiver) SaveDir(ctx context.Context, snPath string, fi os.FileInfo, dir string, previous *restic.Tree) (d FutureTree, err error) {
+func (arch *Archiver) SaveDir(ctx context.Context, snPath string, fi os.FileInfo, dir string, previous *restic.Tree, complete CompleteFunc) (d FutureTree, err error) {
```

Comment:
Could the callback be a field on the Archiver instead?

Reply:
SaveDir misses two parameters used in the callback: `previous` and `time.Since(start)` these are captured in the function closure. The inspiration for the callback was the call to `arch.fileSaver.Save` which also uses a callback of the same type.